proto/documents/v1alpha/web_publishing.proto

```diff
@@ -0,0 +1,330 @@
+syntax = "proto3";
+
+package com.mintter.documents.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+option go_package = "mintter/backend/genproto/documents/v1alpha;documents";
+
+// Service exposed by the local application backend for
+// configuring and managing remote Mintter Sites.
+service WebPublishing {
+  // Adds a site configuration to the local app backend.
+  rpc AddSite(AddSiteRequest) returns (SiteConfig);
+
+  // Removes site configuration from the local app backend.
+  rpc RemoveSite(RemoveSiteRequest) returns (google.protobuf.Empty);
+
+  // Lists configured sites.
+  rpc ListSites(ListSitesRequest) returns (ListSitesResponse);
+
+  // Lists web publication records for a given document among the configured web sites.
+  rpc ListWebPublicationRecords(ListWebPublicationRecordsRequest) returns (ListWebPublicationRecordsResponse);
+}
+
+// Request to add a site.
+message AddSiteRequest {
+  // Required. Site hostname.
+  string hostname = 1;
+
+  // Optional. Invite token for the site. Not needed
+  // if the site already knows our Account ID.
+  string invite_token = 2;
+}
+
+// Request to remove a site from the local app backend.
+message RemoveSiteRequest {
+  // Required. Site hostname.
+  string hostname = 1;
+}
+
+// Request to list configured sites.
+message ListSitesRequest {
+  // Optional. Number of items per page.
+  int32 page_size = 1;
+
+  // Optional. Token for a specific page.
+  string page_token = 2;
+}
+
+// Response with a list of sites.
+message ListSitesResponse {
+  // List of sites.
+  repeated SiteConfig sites = 1;
+
+  // Token for the next page if any.
+  string next_page_token = 2;
+}
+
+// Request to a list publication records for a given document
+// among the configured web sites.
+message ListWebPublicationRecordsRequest {
+  // Required. Document ID.
+  string document_id = 1;
+
+  // Optional. Specific version of a document.
+  string version = 2;
+}
+
+// Response with a list of publication records among the web sites.
+message ListWebPublicationRecordsResponse {
+  repeated WebPublicationRecord publications = 1;
+}
+
+// Local site configuration.
+message SiteConfig {
+  // Hostname of the site.
+  string hostname = 1;
+
+  // Our role on this site.
+  Member.Role role = 2;
```

Comment:
Since `SiteConfig` is provided by the **local call** `ListSites`, the role here may change in the server and the local version of it may change. The proper remote call to get our role from, should be `GetMember` that actually gets the data from the site's database. So do we keep two different records of the role (one locally outdated and one remotely? or we make a remote query under the hood on the local query (`ListSites`)? or we delete the role response from the local query (`ListSites`) in favor of the remote one (`GetMember`)? cc @burdiyan @ericvicenti @horacioh

Reply:
could we add member info to the remote call `GetSiteInfo` ? that aligns well with the current UI needs - we don't currently show the role when listing sites

Reply:
But you already have member information in the `ListMembers` endpoint. Do you want to save one call?

Reply:
True, that is close enough. The front end has to figure out which member in the list is "me", but that is easy enough. 

At the moment I can't think of a reason we need it outside of 'GetMembers'

Reply:
@ericvicenti see [this related thread](https://github.com/mintterteam/mintter/pull/1298/files#r1094714053). If we include the *account_id* filter there you will have what you want also here just calling `Listmembers` and we can get rid of `SiteConfig.role` and `getMember` Altogether